frontend/src/components/problem/ProblemDisplay.tsx

```diff
@@ -142,141 +173,218 @@ function ProblemDisplay(props: ProblemDisplayParams) {
   };
 
   return (
-    <Content>
-      <MediumText>Name:</MediumText>
-      <TextInput
-        name="name"
-        value={newProblem.name}
-        onChange={handleChange}
-      />
+    <>
+      <MainContent>
+        <FlexBareContainer>
+          <SmallHeaderText>Problem</SmallHeaderText>
+          <SmallButtonLink
+            color={ThemeConfig.colors.gradients.gray}
+            onClick={() => onClick(newProblem)}
+            to="/problems/all"
+            style={{ marginLeft: 'auto' }}
+          >
+            Back
+          </SmallButtonLink>
+          <SmallButton
+            color={ThemeConfig.colors.gradients.blue}
+            onClick={() => onClick(newProblem)}
+          >
+            {actionText}
+          </SmallButton>
+        </FlexBareContainer>
+        <SettingsContainer padding="3rem">
+          <PureTextInputTitle
+            placeholder="Write a nice title"
+            name="name"
+            value={newProblem.name}
+            onChange={handleChange}
+          />
+          <hr style={{ margin: '1rem 0' }} />
+          <MarkdownEditor
+            placeholder="Write a nice description"
+            defaultValue={newProblem.description}
+            onChange={(getNewValue) => handleDescriptionChange(getNewValue())}
+          />
+        </SettingsContainer>
 
-      <MediumText>Description:</MediumText>
-      <StyledMarkdownEditor
-        defaultValue={newProblem.description}
-        onChange={(getNewValue) => handleDescriptionChange(getNewValue())}
-      />
+        {editMode
+          ? (
+            <>
+              <SmallHeaderText>Test Cases</SmallHeaderText>
+              {newProblem.testCases.map((testCase, index) => (
+                <SettingsContainer>
+                  <FlexBareContainer>
+                    <MarginRightContainer marginRight="10">
+                      <Text style={{ marginTop: 0 }}>Input</Text>
+                      <FixedTextArea
+                        value={newProblem.testCases[index].input}
+                        onChange={(e) => {
+                          const current = newProblem.testCases[index];
+                          handleTestCaseChange(index, e.target.value,
+                            current.output, current.hidden, current.explanation);
```

Comment:
This might be a later on discussion but will we be using the explanation field anytime in the near future? I'm imagining when adding TeamsCode problems with multiple test cases per input that the explanations might become kind of redundant, like maze problem -> "this is the correct path" or find max -> "X is the max in the array" or palindrome -> "the given string is/is not a palindrome". Although it could be useful down the line with a teacher-oriented route for certain types of beginner problems - that might be a bit far in the future though from what I'm imagining

It's an optional field though so I'm def fine for keeping it as is for now tho 👍

Reply:
Yeah my thinking is to keep it now, as I do think it could be helpful for teachers. There is another idea I actually wanted to bring up - thoughts on having a "name" for each test case? This would be optional, but essentially for any hidden test case, it would allow the user to see the name of the test case and perhaps troubleshoot from there - I think this is how Gradescope does it, for example.

Not a super important feature, but an easy one to add - though potentially tedious for the problem maker. Let me know your thoughts on this (these could be auto-generated too imo, if you have the same set of inputs every time).

Reply:
Hm yeah that could be a good idea and also a good beginner story for new members. I agree though that it would be tedious for the problem maker - the way I'm thinking about it is that depending on the path we go we'd either go all in on name + explanation (e.g. Gradescrope route as you mentioned) or perhaps leave both out (e.g. LeetCode-style contests)

In the meantime though we could also do something with test case # kind of like how LeetCode does, so if you fail one it'll show like 45/90 passed, and then show the 46th test case or something

Reply:
That all sounds good, could definitely be a good first story and we could definitely show #'s for test cases.